client/js/is-sum-function.js

```diff
@@ -0,0 +1,73 @@
+const ofRightForm = function(string) {
+	// if string is of right form, return array of parameters
+	// otherwise return false
+
+	// see if string is of form "= [stuff1] ( [stuff2] : [stuff3] )"
+	// stuff1 contains no "("; stuff2 contains no ":"; stuff3 contains no ")"
+	// stuff1, stuff2, stuff3 must be at least one character
+	const symbolTester = /^=([^\(]+)\(([^:]+):([^\)]+)\)$/
+
+	// parameters is null if string is NOT of appropriate form
+	// otherwise, parameters[1], parameters[2] and parameters[3]
+	// contain the needed parameters
+	const parameters = string.match(symbolTester);
+	
+	if (parameters === null) {
+		return false;
+	} else {
+		return parameters.slice(1, 4); // [stuff1, stuff2, stuff3]
+	}
+}
+
+const cellOfRightForm = function(string) {
+  // see if string is of form "[CapitalLetter][PositiveInteger]"
+  // if the string is NOT of that form, return false
+  // otherwise return array in form [CapitalLetter, PositiveInteger]
+
+  const tester = /^([A-Z]{1})([0-9]+)$/
+  const result = string.match(tester);
+
+  if (result === null) {
+  	return false;
+  } else {
+  	return result.slice(1, 3);
+  }
+}
+
+const isSumFunction = function(string) {
+	// return [startCol, startRow, endRow] if string is a sum function
+	// otherwise return false
+
+	// string needs to be of right form
+	if (ofRightForm(string) !== null) {
```

Comment:
This is way, way, way too much processing just to do this task.

You can use `String.match()`. [MDN link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match)

Here's the regular expression you need:
`=SUM\(([A-Z]\d*):([A-Z]\d*)\)`

It's saying:
1. exact match on `=`
2. exact match on `SUM`
3. escape the left parenthesis using a backslash, so exact match on `(`
4. un-escaped left parenthesis means "start capturing the matched content".
5. [A-Z] means match one character from A to Z
6. `\d*` means match 0 or more digits
7. un-escaped right parenthesis means "stop capturing the matched content". So anything that got matched between those parentheses becomes 1 match group.
8. exact match on `:`
9. same as steps 3 to 7
10. escape the right parenthesis using a backslash, so exact match on `)`

You can test your regular expressions with this handy tool (which is how I built up the above regex, step by step):
http://scriptular.com/

![image](https://cloud.githubusercontent.com/assets/17621133/25673591/cb4b7d48-2fec-11e7-8b48-105f6fdec3d1.png)